openhands_aci/editor/editor.py

```diff
@@ -174,12 +174,30 @@ def view(self, path: Path, view_range: list[int] | None = None) -> CLIResult:
                     'The `view_range` parameter is not allowed when `path` points to a directory.',
                 )
 
+            # First count hidden files/dirs in current directory only
+            # -mindepth 1 excludes . and .. automatically
+            _, hidden_stdout, _ = run_shell_cmd(
+                rf"find -L {path} -mindepth 1 -maxdepth 1 -name '.*'"
+            )
+            hidden_count = (
+                len(hidden_stdout.strip().split('\n')) if hidden_stdout.strip() else 0
+            )
+
+            # Then get non-hidden files/dirs, excluding hidden files at all levels
+            # We need to exclude both files/dirs starting with . and files/dirs under hidden dirs
             _, stdout, stderr = run_shell_cmd(
                 rf"find -L {path} -maxdepth 2 -not -path '*/\.*'",
                 truncate_notice=DIRECTORY_CONTENT_TRUNCATED_NOTICE,
             )
             if not stderr:
-                stdout = f"Here's the files and directories up to 2 levels deep in {path}, excluding hidden items:\n{stdout}\n"
+                msg = [
+                    f"Here's the files and directories up to 2 levels deep in {path}, excluding hidden items:\n{stdout}"
+                ]
+                if hidden_count > 0:
+                    msg.append(
+                        f"\n{hidden_count} hidden files/directories are excluded. You can use 'ls -la {path}' to see them."
```

Comment:
I think we should make it clear the count is for current dir only, because for non-hidden items we include those up to **2** level?

```suggestion
                        f"\n{hidden_count} hidden files/directories in the current directory are excluded. You can use 'ls -la {path}' to see them."
```